jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/JMSHandleConfiguration.java

```diff
@@ -0,0 +1,297 @@
+
+        // Get the general options from the jmsConfig
+        if ( !(jmsConfig.get("general") instanceof Map)) {
+            log.error("Configuration failed. No general options specified.");
+            failConfig();
+            return;
+        }
+        general = (Map) jmsConfig.get("general");
+
+        // Get the sender options from the jmsConfig
+        if ( !(jmsConfig.get("sender") instanceof Map)) {
+            log.error("Configuration failed. No sender options specified.");
+            failConfig();
+            return;
+        }
+        sender = (Map) jmsConfig.get("sender");
+
+        // Get the receiver options from the jmsConfig
+        if ( !(jmsConfig.get("receiver") instanceof Map)) {
+            log.error("Configuration failed. No receiver options specified.");
+            failConfig();
+            return;
+        }
+        receiver = (Map) jmsConfig.get("receiver");
+
+        boolean success = createJMSConnection(general, sender, receiver);
+        if (!success) {
+            failConfig();
+            return;
+        }
+
+        log.trace("Setup complete");
+        configComplete = true;
+    }
+
+    /**
+     * Attempts to create the JMS Connection based on the configuration document.
+     * @param generalConfig     The general configuration for the JMS Source
+     * @param sender            The sender configuration for the JMS Source
+     * @param receiver          The receiver configuration for the JMS Source
+     * @return                  true if the JMS connections could be created, false otherwise
+     */
+    boolean createJMSConnection(Map<String, ?> generalConfig, Map<String, ?> sender, Map<String, ?> receiver) {
+        // Get Provider URL, Connection Factory, and Initial Context from general config
+        String providerURL;
+        String connectionFactory;
+        String initialContext;
+        String username;
+        String password;
+        
+        if (generalConfig.get("providerURL") instanceof String) {
+            providerURL = (String) generalConfig.get("providerURL");
+        } else {
+            log.error("Configuration failed. No providerURL was specified");
+            return false;
+        }
+
+        if (generalConfig.get("connectionFactory") instanceof String) {
+            connectionFactory = (String) generalConfig.get("connectionFactory");
+        } else {
+            log.error("Configuration failed. No connectionFactory was specified");
+            return false;
+        }
+
+        if (generalConfig.get("initialContext") instanceof String) {
+            initialContext = (String) generalConfig.get("initialContext");
+            
+            // Check to make sure we only have one InitialContextFactory
+            String initialContextCheck = source.initialContextMap.get(initialContext);
+            if (initialContextCheck == null) {
+                if (source.initialContextMap.size() > 1) {
```

Comment:
I think the test here should be
```
if (source.initialContextMap.size() > 0) {
```

( or `!= 0`). That is, we don't have one with this name already, so we're going to need to add it.  Once we add it, will we be over the limit of "1" -- meaning we were at zero before that.

Even easier might be change the map to a string, and if it's null or matches the requested one, then things are OK. Otherwise, not.

Reply:
I am going to change it to a string since that definitely seems like the simplest solution here, sorry for the silly mistake.